cli/cmd/dashboard.go

```diff
@@ -55,6 +74,26 @@ var dashboardCmd = &cobra.Command{
 	},
 }
 
+func isDashboardAvailable(kubeAPI k8s.KubernetesApi) (bool, error) {
+	client, err := kubeAPI.NewClient()
```

Comment:
If we keep the URL generation here, one pattern we've been using across the CLI is to make sure that all [service objects](https://books.google.com/books?id=xColAAPGubgC&lpg=PP1&dq=domain%20driven%20design%20service&pg=PA105#v=onepage&q=service&f=false) are instantiated in the command `runE` block and passed down to all functions. This is an application of the [dependency inversion principle](https://web.archive.org/web/20150905081103/http://www.objectmentor.com/resources/articles/dip.pdf) that allows for us to test these functions without much hassle.

Reply:
Hmm, just to clarify, are you suggesting I move the kubeAPI.NewClient into the run block and therefore only send an `http.Client` to the `isDashboardAvailable` function?

Reply:
It depends, tbh

If we were to keep the current design, I'd suggest we change this func's signature to `func isDashboardAvailable(kubeClient k8s.K8sClient, kubeAPI k8s.KubernetesApi) (bool, error)`.

If we are going ahead with changing this function to only check the url, I think the signature should be `func isDashboardAvailable(dashboardUrl string) (bool, error)`, and it's ok to contain any `http` code within this func's body.

Reply:
Sort of related: Perhaps we should re-use the code that `conduit check` uses to verify conduit is installed:
https://github.com/runconduit/conduit/blob/9a40d984ff75ef16c14e9e2e8af567f7dc38c6a2/cli/cmd/check.go#L44

There's additional logic there to handle if we're an internal or external client:
https://github.com/runconduit/conduit/blob/9a40d984ff75ef16c14e9e2e8af567f7dc38c6a2/controller/api/public/client.go#L180